imperial_coldfront_plugin/models.py

```diff
@@ -1 +1,28 @@
 """Plugin Django models."""
+
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.db import models
+
+
+class GroupMember(models.Model):
+    """Membership relationship within a group, associating an owner with a member.
+
+    This model stores relationships where each instance represents an ownership
+    connection between two users within a group, where `owner` is the user who
+    owns the membership (or perhaps manages the group) and `member` is the
+    associated user.
+
+    Attributes:
+        owner (ForeignKey): A reference to the user designated as owner, connected to
+            AUTH_USER_MODEL. Deletes related memberships when the owner is deleted.
+        member (ForeignKey): A reference to the user designated as member, connected to
+            AUTH_USER_MODEL. Deletes related memberships when the member is deleted.
+    """
+
+    owner: User = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
+    member: User = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
```

Comment:
I believe you can get around the mypy error by removing the `User` annotation on the variable. The annotation is incorect as the attribute is not an instance of `User` but an instance of `ForeignKey` which is mapped to `User`.

Reply:
So mypy embraces a sort of minimal approach to typing where it will automatically infer as much type information as possible - see https://mypy.readthedocs.io/en/stable/type_inference_and_annotations.html for details. You _can_ add explicit annotations as well but I'd usually hesitate to do that because:
- I think it detracts from readability rather than adding because it adds clutter and redundant information to the codebase. Mypy will tell you if it needs additional info to work out the type of something. It may sound odd but I think the exact type of something is usually a secondary consideration if using mypy to check that everything is consistent.
- Explicitly typing everywhere makes it difficult to update or refactor later.
- A good ide can provide you the type information on demand on those occasions it's required.